mkosi/distributions/gentoo.py

```diff
@@ -449,16 +341,39 @@ def mkosi_conf(self) -> None:
                 f"""\
                 # MKOSI: these were used during image creation...
                 # and some more! see under package.*/
-                #
-                # usrmerge (see all under profile/)
                 {emerge_vars_str}
                 """
             )
         )
 
+    def update_stage3(self) -> None:
+        self.invoke_emerge(opts=self.EMERGE_UPDATE_OPTS, pkgs=self.pkgs['boot'])
+        self.invoke_emerge(opts=["--config"],
+                           pkgs=["sys-kernel/gentoo-kernel-bin"])
+        self.invoke_emerge(opts=self.EMERGE_UPDATE_OPTS,
+                           pkgs=self.pkgs['sys'])
+
+    def depclean(self) -> None:
+        self.invoke_emerge(actions=["--depclean"])
+
+    def finishup(self) -> None:
+        self.invoke_emerge(pkgs=self.pkgs['fs'])
+
+        if not self.state.do_run_build_script:
+            if self.state.config.ssh:
+                MkosiPrinter.print_step("enabling sshd")
+                cmd = ["/usr/bin/systemctl", "enable", "sshd"]
+                run_workspace_command(self.state, cmd, nspawn_params=self.DEFAULT_NSPAWN_PARAMS,
+                                      check=True)
```

Comment:
This should be done with host systemctl instead of image systemctl, so use `run()` instead of `run_workspace_command()`

Reply:
Also, why do this here and not in the shared code we have for this?

Reply:
don't remember why. i had a bit in `mkosi/__init__.py` related to this which got removed after switching to usrmerge. removed it to see what broke in the past, although it seems to me that CI doesn't have a test for ssh?

Reply:
seems fine, closing this.